packages/wrangler/src/r2/lock.ts

```diff
@@ -0,0 +1,333 @@
+
+		if (rules.length === 0) {
+			logger.log(`There are no lock rules for bucket '${bucket}'.`);
+		} else {
+			const tableOutput = tableFromBucketLockRulesResponse(rules);
+			logger.log(tableOutput.map((x) => formatLabelledValues(x)).join("\n\n"));
+		}
+	},
+});
+
+export const r2BucketLockAddCommand = createCommand({
+	metadata: {
+		description: "Add a lock rule to an R2 bucket",
+		status: "stable",
+		owner: "Product: R2",
+	},
+	positionalArgs: ["bucket", "id", "prefix"],
+	args: {
+		bucket: {
+			describe: "The name of the R2 bucket to add a bucket lock rule to",
+			type: "string",
+			demandOption: true,
+		},
+		id: {
+			describe: "A unique identifier for the bucket lock rule",
+			type: "string",
+			requiresArg: true,
+		},
+		prefix: {
+			describe:
+				'Prefix condition for the bucket lock rule (set to "" for all prefixes)',
+			type: "string",
+			requiresArg: true,
+		},
+		"lock-days": {
+			describe: "Number of days after which objects expire",
+			type: "string",
+			conflicts: "lock-date",
+		},
+		"lock-date": {
+			describe: "Date after which objects expire (YYYY-MM-DD)",
+			type: "string",
+			conflicts: "lock-days",
+		},
+		jurisdiction: {
+			describe: "The jurisdiction where the bucket exists",
+			alias: "J",
+			requiresArg: true,
+			type: "string",
+		},
+		force: {
+			describe: "Skip confirmation",
+			type: "boolean",
+			alias: "y",
+			default: false,
+		},
+	},
+	async handler(
+		{ bucket, lockDays, lockDate, jurisdiction, force, id, prefix },
+		{ config }
+	) {
+		const accountId = await requireAuth(config);
+
+		const rules = await getBucketLockRules(accountId, bucket, jurisdiction);
+
+		if (!id && !isNonInteractiveOrCI() && !force) {
+			id = await prompt("Enter a unique identifier for the lock rule");
+		}
+
+		if (!id) {
+			throw new UserError("Must specify a rule ID.", {
+				telemetryMessage: true,
+			});
+		}
+
+		const newRule: BucketLockRule = {
+			id: id,
+			enabled: true,
+			condition: { type: "Indefinite" },
+		};
+		if (prefix === undefined && !force) {
+			prefix = await prompt(
+				'Enter a prefix for the bucket lock rule (set to "" for all prefixes)',
+				{ defaultValue: "" }
+			);
+			if (prefix === undefined) {
+				const confirmedAdd = await confirm(
+					`Are you sure you want to add lock rule '${id}' to bucket '${bucket}' without a prefix? ` +
+						`The lock rule will apply to all objects in your bucket.`
+				);
+				if (!confirmedAdd) {
+					logger.log("Add cancelled.");
+					return;
+				}
+			}
+		}
+
+		if (prefix) {
+			newRule.prefix = prefix;
+		}
+
+		if (lockDays !== undefined) {
+			if (isNonNegativeNumber(lockDays)) {
+				const conditionDaysValue = Number(lockDays) * 86400; // Convert days to seconds
+				newRule.condition = {
+					type: "Age",
+					maxAgeSeconds: conditionDaysValue,
+				};
+			} else {
+				throw new UserError(`Must be a positive number: ${String(lockDays)}`, {
+					telemetryMessage: true,
```

Comment:
```suggestion
					telemetryMessage:{lock days not a positive number},
```
(easier for us to aggregate if we don't include the actual number)